include/z64light.h

```diff
@@ -60,6 +72,52 @@ typedef enum {
     /* 0x02 */ LIGHT_POINT_GLOW
 } LightType;
 
-typedef void (*LightsBindFunc)(Lights* lights, LightParams* params, Vec3f* vec);
+typedef void (*LightsBindFunc)(Lights* lights, LightParams* params, Vec3f* objPos);
+
+// Helper for defining functions that have extra file/line args in debug vs non-debug
+#if IS_DEBUG
+#define LIGHTS_DEBUG_ARGS(file, line) , file, line
+#else
+#define LIGHTS_DEBUG_ARGS(file, line)
+#endif
+
+//  Light Info
+void Lights_PointSetInfo(LightInfo* info, s16 x, s16 y, s16 z, u8 r, u8 g, u8 b, s16 radius, s32 type);
+void Lights_PointNoGlowSetInfo(LightInfo* info, s16 x, s16 y, s16 z, u8 r, u8 g, u8 b, s16 radius);
+void Lights_PointGlowSetInfo(LightInfo* info, s16 x, s16 y, s16 z, u8 r, u8 g, u8 b, s16 radius);
+void Lights_PointSetColorAndRadius(LightInfo* info, u8 r, u8 g, u8 b, s16 radius);
+void Lights_PointSetPosition(LightInfo* info, s16 x, s16 y, s16 z);
+void Lights_PointSetAttenuation(LightInfo* info, u8 kc, u8 kl, u8 kq);
+void Lights_PointSetSpecularSize(LightInfo* info, u8 size);
+void Lights_DirectionalSetInfo(LightInfo* info, s8 x, s8 y, s8 z, u8 r, u8 g, u8 b);
+void Lights_DirectionalSetSpecularSize(LightInfo* info, u8 size);
+
+//  Light Context
+void LightContext_Init(struct PlayState* play, LightContext* lightCtx);
+void LightContext_DestroyList(struct PlayState* play, LightContext* lightCtx);
```

Comment:
This is not the case for upstream and shouldn't be the case here either. Eventually functions.h will cease to exist as more work is done upstream to move functions and variables to their respective header files. Since I removed and added a bunch of functions, I felt I should just move these now.

Reply:
I think PRs to this project should stick to the code standards of the project currently, not the code standards that upstream may be moving to in the future. Personally I think putting functions with each subsystem is nicer organization, but I also think writing `struct PlayState` everywhere is ugly and a major regression in terms of how modern the code looks. But anyway I will not block the PR over this.

Reply:
Some headers in this project have already adopted these changes as they're being rolled out incrementally upstream. See `irqmgr.h`, `sched.h`, `padmgr.h`, `tha.h`, `thga.h`, and probably others (these are just the ones I've already done this for upstream). There's little point in waiting for upstream in this case, it's eventually going to cover all headers in this way.